### PR TITLE
Zabbix API util

### LINKED ISSUE #15: Create reusable Zabbix api call utility
Subsequent tasks will require making Zabbix api calls. So far, we have an implementation [here](https://github.com/kosyfrances/rundeck_zabbix/blob/a5b9db47919c536ed2394c78af4eb7a6efdb4bff/zabbix_api_key.go) where we are making a call to Zabbix API to get the api key.  We can probably have a `utils` package when/if the project gets bigger. 

In this current issue, we want to accomplish the following:
- That zabbix_api_key.go file is converted to a utils.go file
- That we have a generic Zabbix api call function
- That the current functionality to get api key is derived from the generic Zabbix api call function

### PR DESCRIPTION
**Pull request by kosyfrances**
This PR does the following:

* Moves `zabbix_api_key.go` to `utils/zabbix/utils.go`
* Moves `zabbix_api_key_test.go` to `utils/zabbix/utils_test.go`
* Refactors existing code to create reusable Zabbix API utilities
* Updates Readme to reflect current Zabbix API version being used

Fixes #15

### CODE REVIEW

File: utils/zabbix/utils.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		log.Fatalf("Cannot make HTTP request. Error: %v\n", err)
+	}
+
+	req.Header.Set("Content-type", "application/json")
+	// Send the request via a client
+	resp, err = http.DefaultClient.Do(req)
+	return resp, err
+}
+
+func (z *ZabbixAPI) getAPIKey() (string, error) {
```
Comment (osule) [inline utils/zabbix/utils.go]: What do you think about checking the struct field for the API Key and returning it if it is set?

File: utils/zabbix/utils.go [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+	err = json.NewDecoder(resp.Body).Decode(&r)
+	if err != nil {
+		return "", fmt.Errorf("cannot decode response. Error: %v", err)
+	}
+	return r.Key, nil
+}
+
+// SetKey in ZabbixAPI Struct and environment(if not already set)
+func (z *ZabbixAPI) SetKey() {
```
Comment (osule) [inline utils/zabbix/utils.go]: I suggest to name this SetAPIKey to be consistent with the getter method.
Does it make sense to set API key in the application environment when the struct is Initialize?
How about minimizing surprise by passing the API Key as an argument to the SetKey method?

**Comment by kosyfrances** [comment]
Thanks for the review @osule. I will be revamping this PR to use configuration values from a file gotten from the initial user setup (see https://github.com/kosyfrances/rundeck-zabbix/issues/19) instead of env vars.

**Comment by kosyfrances** [comment]
@osule @tundeaoni this is ready for another review. We will be getting the configuration values from a config file, and I want to keep the Zabbix API util package as specific as possible, so I think there is no need checking whether values are set or not in this case.

File: utils/zabbix/utils.go (hunk 1 above)
Comment (kosyfrances) [reply utils/zabbix/utils.go]: good idea. will do that.

File: utils/zabbix/utils.go (hunk 2 above)
Comment (kosyfrances) [reply utils/zabbix/utils.go]: Removed the function entirely.

File: utils/zabbix/utils.go [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+type ZabbixAPI struct {
+	URL      string
+	Key      string
+	User     string
+	Password string
+}
+
+// MakeAPIRequest makes an API call to Zabbix.
+// It returns a response object and an error object.
+func (z *ZabbixAPI) MakeAPIRequest(payload map[string]interface{}) (resp *http.Response, err error) {
```
[the remark this answers is not in the document]
Comment (kosyfrances) [reply utils/zabbix/utils.go]: @osule we can make this explicit when the need arises. As discussed offline, I will be going on to merge this PR.

**Pull request merged**